.github/workflows/integration-tests.yaml

```diff
@@ -20,29 +20,34 @@ on:
 
 jobs:
   wait-for-image-build:
+    environment: ${{ github.event.pull_request.author_association != 'COLLABORATOR' && github.event.pull_request.author_association != 'OWNER' && 'restricted' || 'internal' }}
```

Comment:
I would ad the justification why we to guard this as a comment to the environment instead of inline at the action. Also `# This is dangerous without the environment check` doesn't explain why. I guess because the a GH token with extended permissions is used?